tests/io/eager/test_commit.py

```diff
@@ -302,3 +299,35 @@ def test_commit_dataset_delete_all(store, metadata_version):
     )
     assert len(updated_dataset.partitions) == 0
     assert updated_dataset.explicit_partitions is True
+
+
+def test_commit_dataset_add_metadata(store, metadata_version):
+    df = pd.DataFrame({"p": [1]})
+    dataset = store_dataframes_as_dataset(
+        dfs=[df],
+        store=lambda: store,
+        metadata={"old": "metadata"},
+        dataset_uuid="dataset_uuid",
+        metadata_version=metadata_version,
+    )
+
+    commit_dataset(store=store, dataset_uuid=dataset.uuid, metadata={"new": "metadata"})
+    restored_ds = DatasetMetadata.load_from_store(uuid="dataset_uuid", store=store)
+
+    actual_metadata = restored_ds.metadata
+    del actual_metadata["creation_time"]
+    expected_meta = {"new": "metadata", "old": "metadata"}
+    assert actual_metadata == expected_meta
+
```

Comment:
```suggestion
def test_commit_dataset_add_metadata(store, metadata_version):
    df = pd.DataFrame({"p": [1]})
    dataset = store_dataframes_as_dataset(
        dfs=[df],
        store=lambda: store,
        metadata={"old": "metadata", "something": "original"},
        dataset_uuid="dataset_uuid",
        metadata_version=metadata_version,
    )

    commit_dataset(store=store, dataset_uuid=dataset.uuid, metadata={"new": "metadata", "something": "fancy"})
    restored_ds = DatasetMetadata.load_from_store(uuid="dataset_uuid", store=store)

    actual_metadata = restored_ds.metadata
    del actual_metadata["creation_time"]
    expected_meta = {"new": "metadata", "old": "metadata",  "something": "fancy"}
    assert actual_metadata == expected_meta
```

would be nice to check that we can modify metadata (if this is supported)

Reply:
Yes this is supported and this is what the metadata merger is all about. I would argue to not mix things here. We should not test every feature in the high level tests imo

Reply:
Testing all low level features on this high level is something we do for a lot of pipelines and I argue this makes the entire test suite very heavy

Reply:
Alright